src/test/java/com/ning/http/client/providers/grizzly/GrizzlyAsyncHttpProviderTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package com.ning.http.client.providers.grizzly;
+
+import static java.lang.Thread.currentThread;
+import static org.mockito.Mockito.mock;
+import static org.testng.Assert.assertEquals;
+
+import com.ning.http.client.AsyncCompletionHandler;
+import com.ning.http.client.AsyncHttpClient;
+import com.ning.http.client.AsyncHttpClientConfig;
+import com.ning.http.client.Request;
+import com.ning.http.client.RequestBuilder;
+import com.ning.http.client.Response;
+import com.ning.http.client.async.AbstractBasicTest;
+import com.ning.http.client.async.ProviderUtil;
+
+import java.util.Map;
+import java.util.concurrent.Future;
+
+import org.slf4j.MDC;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class GrizzlyAsyncHttpProviderTest extends AbstractBasicTest {
+
+  @Test(groups = { "standalone", "default_provider", "async" })
+  public void asyncProviderPreservesClassLoader() throws Throwable {
+    try (AsyncHttpClient client = getAsyncHttpClient(null)) {
```

Comment:
If we want to preserve compatibility for grizzly for JDK<7, remove this try-with-resources.

Reply:
Maybe not, this is a test.